text/0000-deprecate-http-tx.md

```diff
@@ -0,0 +1,109 @@
+
+Previously, some services like exchanges _only_ supported HTTP(S) Grin transactions. This change impacts these services significantly.
+
+Tor is the closest replacement to HTTP(S), but for some services the use of Tor is unacceptable. These services will will need to refactor the user flow to allow supporting armored slate strings instead, as the HTTP(S) method is no longer supported by default for `grin-wallet`.
+
+This up-front cost should be worth the reduced complexity required in supporting the alternative transaction methods as well as the privacy benefits provided to the users of the services.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The technical and implementation details for this RFC are simple: no new features are added and some existing features are restricted:
+
+- `grin-wallet` http listener only accepts connections from localhost
+
+- `grin-wallet listen` no longer accepts `-e` argument
+
+- `grin-wallet send -d` provides a strong warning if the destination is not a valid .onion address
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Forces changes to existing transaction building workflows that use HTTP(S)
+
+- Remaining transaction options (Tor, armored slate, file) may not be appealing for services already exclusively using HTTP(S)
+
+- Forces "always on" wallet listeners to use Tor or to build out logic to support handling armored slates in the UX flow
+
+- Potentially makes a technology that is already challenging for services to adopt even more challenging to adopt
+
+- Those that do not value privacy as a first-order objective may see this as an unnecessary inconvenience
+
+- Friction could cause some services to avoid Grin completely rather than adding support for a new transaction method
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Instead of looking for a best possible transaction method, this proposal comes down to favoring the least worst transaction methods by removing the worst transaction methods from the perspectives of privacy, security, censorship resistance and minimalism.
+
+In this case, HTTP(S) is the least desirable transaction method. It actively jeopardizes user privacy when services only accept payments via HTTP(S), can jeapardize security when not configured properly, is not censorship resistant as IPs can be blocked and domains can be seized and is not minimal once the additional configuration requirements are factored in.
+
+By not deprecating HTTP(S) we lose an opportunity to provide an additional layer of privacy to present and future Grin supporters. While there will likely be a fair amount of friction to the adoption of this RFC, most will agree it is something that will need to be done eventually and will only become more difficult to do with time.
```

Comment:
Not sure this is required here. 

Would be more interesting to understand what other alternatives we could try rather than deprecating https, like for example only deprecating http and so on.

Reply:
Pruned this comment down a bit but wanted to leave a note about what the cost was for not deprecating.